src/banman.cpp

```diff
@@ -159,12 +135,43 @@ bool BanMan::Unban(const CSubNet& sub_net)
     return true;
 }
 
-void BanMan::GetBanned(banmap_t& banmap)
+void BanMan::UnbanAll(const CNetAddr& net_addr)
+{
+    {
+        LOCK(m_cs_banned);
+        std::vector<CSubNet> entries_to_delete;
+        for (const auto& it : m_banned) {
+            CSubNet sub_net = it.first;
+            if (sub_net.Match(net_addr)) entries_to_delete.push_back(sub_net);
+        }
+
+        if (!entries_to_delete.empty()) m_is_dirty = true;
+        for (const auto&it : entries_to_delete) {
+            m_banned.erase(it);
+        }
```

Comment:
Why not erase as you go and use `m_banned.size()` to infer `m_is_dirty`, instead of constructing a whole new vector?

Reply:
Using erase-remove_if is also more efficient - you'll only need to pass over `m_banned` once instead of up to `entries_to_delete.size()` times.

Reply:
yeah `erase_if` would be the best but it's C++20 😭

Reply:
Since `map.erase(it)` invalidates `it` I was doing a double-pass to avoid incrementing `it`. However, it turns out `erase` will [return the Iterator](https://en.cppreference.com/w/cpp/container/map/erase) following the last removed element. So I've updated the code to use that.

@jnewbery I am confused about this:
> you'll only need to pass over m_banned once instead of up to entries_to_delete.size() times.

I think the code is only passing over `m_banned` once. Am I missing something? I also tried erase-remove_if but it does not seem to work for associative containers. Is there an example I can learn from?

Reply:
Sorry. I mistakenly thought that `m_banned` was a vector.